bin/bootstrap.sh

```diff
@@ -47,7 +50,7 @@ if [ "${OTOMI_PULLSECRET-}" != "" ]; then
   cp -f $PWD/docker-compose.yml $ENV_DIR/
   cp -f $PWD/bin/console.sh $bin_path
 fi
-if [ ! $has_otomi ]; then
```

Comment:
What is wrong with detecting if something was set or not in bash? It is a great feature. It might need -z but I propose we don't set unneededly

Reply:
set -u

Reply:
Variables should be quoted in order to avoid word splitting and globing.
Leaving it without quote can lead to security issue.

Reply:
Why insecure? What is the attack vector? Who is the actor? Why is this relevant in this context?

Reply:
And set -u can still be worked around with `${bla-}`

Reply:
ok

Reply:
> Why insecure? What is the attack vector? Who is the actor? Why is this relevant in this context?

it relates to undefined variables that can be overwritten from the outside.

Reply:
There is no "vulnerability" as such

Reply:
I did some research about `true` and `false` commands and it turns out that your usage is incorrect.

```
var1=true
var2=false

if [ ! $var1 ]; then
echo "T"
else
echo "F"
fi

if [ ! $var2 ]; then
echo "T"
else
echo "F"
fi
```
In both cases the "F" string is printed.

Why?
The `[ $var ]` check only if variable exists not a "boolean" value. In fact there is no boolean value in bash.

The proper usage of true/false concept is
```
if ! $var1 ; then
echo "T"
else
echo "F"
fi
```
In this case `$var` is evaluated to command called `true` and is performed.

I strongly discourage you to use this approach, as it is misleading and can make code vulnerable if it is used for variables that comes from environment. Again we do not want environment variables to be evaluated to any shell commands.

Having that my recommendations are as follows:
- booleans
Use  'true' and 'false' strings and always compare them as strings
```
[[ "$var" == "false" ]]
```
- environment variables 
Treat them as strings "$VAR" to prevent globbing wordsplitting.

Last but not least. I would like to propose to follow this guideline: https://github.com/google/styleguide/blob/gh-pages/shellguide.md, with additional remark related to boolean values.
 
References:
https://stackoverflow.com/questions/2953646/how-can-i-declare-and-use-boolean-variables-in-a-shell-script
https://github.com/google/styleguide/blob/gh-pages/shellguide.md

Reply:
True, yet another way and that's the issue with bash - many ways for doing exactly the same thing  .... ehhh

Anyway the google guideline is the following: [[ … ]] is preferred over [ … ]
Let's stick to that.

Reply:
done